.github/workflows/test.yml

```diff
@@ -43,6 +43,9 @@ jobs:
     - name: Execute Python tests
       run: tox
 
+    - name: Execute Python tests with low level chaining API
+      run: tox -e chaining
```

Comment:
it might be better not to introduce another test in the yaml but to embed the test execution in the context

Reply:
you could introduce an `executor` keyword in the context that takes in two options - `cli` and `api` - similar to `checker`

Reply:
the tests will go against the high level API and not the low level API but hopefully that's a simple enough change

Reply:
done, added lower level api as an executor for now, can switch to higher level later once the changes for UX are in.